pkg/kubelet/cm/topologymanager/policy.go

```diff
@@ -147,19 +174,18 @@ func compareHints(bestNonPreferredAffinityCount int, current *TopologyHint, cand
 
 	// If the current bestHint is preferred and the candidate hint is
 	// non-preferred, never update the bestHint, regardless of the
-	// candidate hint's narowness.
+	// candidate hint's fitness.
 	if current.Preferred && !candidate.Preferred {
 		return current
 	}
 
 	// If the current bestHint and the candidate hint are both preferred,
-	// then only consider candidate hints that have a narrower
-	// NUMANodeAffinity than the NUMANodeAffinity in the current bestHint.
+	// then only consider fitter NUMANodeAffinity
 	if current.Preferred && candidate.Preferred {
-		if candidate.NUMANodeAffinity.IsNarrowerThan(current.NUMANodeAffinity) {
-			return candidate
+		return &TopologyHint{
+			NUMANodeAffinity: m.CompareNUMAAffinityMasks(candidate.NUMANodeAffinity, current.NUMANodeAffinity),
+			Preferred:        true,
```

Comment:
Hmm, maybe we should define a new function hanging off of `HintMerger` as
```
func (m HintMerger) GetBetterHint(h1, h2 *TopologyHint) *TopologyHint
```
which then uses `CompareNUMAAffinityMasks` under the hood

That way, these calls can just turn directly into:
```
return m.GetBetterHint(candidate, current)
```